pkg/db/tx.go

```diff
@@ -128,18 +120,28 @@ func (d *dbTx) Exec(query string, args ...interface{}) (pgconn.CommandTag, error
 		"query": queryToString(query),
 		"took":  time.Since(start),
 	})
-	if isUniqueViolation(err) {
-		return nil, ErrAlreadyExists
-	}
 	if err != nil {
-		dbErrorsCounter.WithLabelValues("exec").Inc()
-		log.WithError(err).Error("SQL query failed with error")
-		return res, err
+		return res, d.handleSQLError(err, "exec", query)
 	}
 	log.Trace("SQL query executed successfully")
 	return res, err
 }
 
+func (d *dbTx) handleSQLError(err error, cmdType string, query string) error {
+	d.logger.Trace("SQL operation err:", err)
+	// Each error that is added here should be updated also in controller.go:handleAPIError
+	if isUniqueViolation(err) {
+		return ErrAlreadyExists
+	}
+	if pgxscan.NotFound(err) {
+		return ErrNotFound
+	}
+
+	dbErrorsCounter.WithLabelValues(cmdType).Inc()
+	d.logger.WithError(err).Error("SQL operation unhandled error")
```

Comment:
Same as in line 131 - WithError and log additional fields.
About the message - "unhandled" is something we don't know yet - prefer to report that the operation completed with error

Reply:
Decided to pass the logger to handleSQLError not lose context, it is now no longer a member function anymore. Let me know if I should move it back to errors.go